tests/api/testcl.hpp

```diff
@@ -107,7 +107,7 @@ struct holder {
             deleter();
         }
     }
-    void deleter() = delete;
```

Comment:
What was the issue with `delete`?

Reply:
Clang does not seem to like calling a deleted method in a template class even when there is a specialized (non-deleted) definition of that method for all instantiations of the template. 

There has been similar issues in the past that have now been fixed, e.g.:
https://bugs.llvm.org/show_bug.cgi?id=17537
But I guess they haven't hit the case where the deleted method is a member of templated class before.

Reply:
Right, I had a feeling this was a case of broken toolchain. Thanks for the details (and sorry I didn't spot the commit message that described the issue before I asked).